pkg/vz/vm_darwin.go

```diff
@@ -0,0 +1,485 @@
+		if err != nil {
+			return err
+		}
+		config.SetDirectoryShare(share)
+		mounts[i] = config
+	}
+	vmConfig.SetDirectorySharingDevicesVirtualMachineConfiguration(mounts)
+	return nil
+}
+
+func attachOtherDevices(_ *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	entropyConfig, err := vz.NewVirtioEntropyDeviceConfiguration()
+	if err != nil {
+		return err
+	}
+	vmConfig.SetEntropyDevicesVirtualMachineConfiguration([]*vz.VirtioEntropyDeviceConfiguration{
+		entropyConfig,
+	})
+
+	configuration, err := vz.NewVirtioTraditionalMemoryBalloonDeviceConfiguration()
+	if err != nil {
+		return err
+	}
+	vmConfig.SetMemoryBalloonDevicesVirtualMachineConfiguration([]vz.MemoryBalloonDeviceConfiguration{
+		configuration,
+	})
+
+	deviceConfiguration, err := vz.NewVirtioSocketDeviceConfiguration()
+	vmConfig.SetSocketDevicesVirtualMachineConfiguration([]vz.SocketDeviceConfiguration{
+		deviceConfiguration,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Set audio device
+	inputAudioDeviceConfig, err := vz.NewVirtioSoundDeviceConfiguration()
+	if err != nil {
+		return err
+	}
+	inputStream, err := vz.NewVirtioSoundDeviceHostInputStreamConfiguration()
+	if err != nil {
+		return err
+	}
+	inputAudioDeviceConfig.SetStreams(
+		inputStream,
+	)
+
+	outputAudioDeviceConfig, err := vz.NewVirtioSoundDeviceConfiguration()
+	if err != nil {
+		return err
+	}
+	outputStream, err := vz.NewVirtioSoundDeviceHostOutputStreamConfiguration()
+	if err != nil {
+		return err
+	}
+	outputAudioDeviceConfig.SetStreams(
+		outputStream,
+	)
+	vmConfig.SetAudioDevicesVirtualMachineConfiguration([]vz.AudioDeviceConfiguration{
+		inputAudioDeviceConfig,
+		outputAudioDeviceConfig,
+	})
+
+	// Set pointing device
+	pointingDeviceConfig, err := vz.NewUSBScreenCoordinatePointingDeviceConfiguration()
+	if err != nil {
+		return err
+	}
+	vmConfig.SetPointingDevicesVirtualMachineConfiguration([]vz.PointingDeviceConfiguration{
+		pointingDeviceConfig,
+	})
+
+	// Set keyboard device
+	keyboardDeviceConfig, err := vz.NewUSBKeyboardConfiguration()
+	if err != nil {
+		return err
+	}
+	vmConfig.SetKeyboardsVirtualMachineConfiguration([]vz.KeyboardConfiguration{
+		keyboardDeviceConfig,
+	})
+	return nil
+}
+
+func getMachineIdentifier(driver *driver.BaseDriver) (*vz.GenericMachineIdentifier, error) {
+	identifier := filepath.Join(driver.Instance.Dir, filenames.VzIdentifier)
+	if _, err := os.Open(identifier); os.IsNotExist(err) {
+		machineIdentifier, err := vz.NewGenericMachineIdentifier()
+		if err != nil {
+			return nil, err
+		}
+		err = os.WriteFile(identifier, machineIdentifier.DataRepresentation(), 0666)
+		if err != nil {
+			return nil, err
+		}
+		return machineIdentifier, nil
+	}
+	return vz.NewGenericMachineIdentifierWithDataPath(identifier)
+}
+
+func getEFI(driver *driver.BaseDriver) (*vz.EFIVariableStore, error) {
+	efi := filepath.Join(driver.Instance.Dir, filenames.VzEfi)
+	if _, err := os.Open(efi); os.IsNotExist(err) {
```

Comment:
I think [Stat](https://pkg.go.dev/os#Stat) is better because if the file exists, it will not be closed until it is GC'd.

Reply:
Done